### DISCUSSION TITLE
How can a Blazor server side app redirect from http://hostname to http://hostname/route1 on load?

### DISCUSSION THREAD

**Discussion by mhsimkin** [dotnet/aspnetcore#29697] [Q&A]
Hi:

For several of the sites I have built using MVC/TypeScript or MVC/Angular I have needed to redirect the user immediately from http://hostname to http://hostname/route1.  I am trying to figure out how to implement the same functionality with Blazor Server Side.

I am aware that this can be done using the IIS URL Rewrite engine, however, it has not been determined if the application will be hosted under IIS, self-hosted on Windows, or hosted on Linux.

Is this possible? If so, what is the best practice approach for this implementation?

Thanks

**Comment by alextech** [comment]
Use `NavigationManager.NavigateTo("url", true)` as early as possible in your main app lifecycle (usually OnInitialized()). Second parameter set to true forces full browser navigation towards it.
